origin-discovery/src/apollo/db.js

```diff
@@ -88,4 +88,66 @@ async function getListing (listingId) {
   return listing
 }
 
-module.exports = { getListing, getListingsById, getListingsBySeller }
+/**
+ * Helper function. Returns an offer object compatible with the GraphQL Offer schema.
+ * @param {Object} row - Row read from DB offer table.
+ * @return {Object}
+ * @private
+ */
+function _makeOffer (row) {
+  return {
+    id: row.id,
+    ipfsHash: row.data.ipfs.hash,
+    data: row.data,
+    status: row.status,
+    price: row.data.price
+  }
+}
+
+/**
+ * Queries DB to get offers.
+ * @param {string} listingId - optional listing id
+ * @param {string} buyerAddress - optional buyer address
+ * @param {string} sellerAddress - optional seller address
+ * @return {Promise<Array<Offer>>}
+ */
+async function searchOffers (listingId, buyerAddress, sellerAddress) {
```

Comment:
nit:
 - the name search Offer makes me thing this could be using ElasticSearch :) perhaps we could rename this simply "getOffers" ?
 - consider passing an object "opts" with fields listingId, buyerAddress, sellerAddress rather than optional arguments. I personally alway find that JS is confusing when it comes to its handling of optional arguments since unlike Python it only considers the position of the arguments but not their name. For example searchOffers(buyerAddress="0x...") would fail because it would use that address as a listing Id. But maybe it is just me :)

Reply:
Thanks good point